src/arch/amdgpu/vega/insts/mubuf.cc

```diff
@@ -823,6 +823,209 @@ namespace VegaISA
     Inst_MUBUF__BUFFER_LOAD_SSHORT::completeAcc(GPUDynInstPtr gpuDynInst)
     {
     } // execute
+    // --- Inst_MUBUF__BUFFER_LOAD_SHORT_D16 class methods ---
+
+    Inst_MUBUF__BUFFER_LOAD_SHORT_D16
+        ::Inst_MUBUF__BUFFER_LOAD_SHORT_D16(InFmt_MUBUF *iFmt)
+        : Inst_MUBUF(iFmt, "buffer_load_short_d16")
+    {
+        setFlag(MemoryRef);
+        setFlag(Load);
+        if (instData.LDS) {
+            setFlag(GroupSegment);
+            warn("BUFFER.LDS not implemented!");
+        } else {
+            setFlag(GlobalSegment);
+        }
+    } // Inst_MUBUF__BUFFER_LOAD_SHORT_D16
+
+    Inst_MUBUF__BUFFER_LOAD_SHORT_D16::~Inst_MUBUF__BUFFER_LOAD_SHORT_D16()
+    {
+    } // ~Inst_MUBUF__BUFFER_LOAD_SHORT_D16
+
+    // --- description from .arch file ---
+    // RETURN_DATA[15 : 0].u16 = MEM[ADDR].u16;
+    // // RETURN_DATA[31:16] is preserved.
+    void
+    Inst_MUBUF__BUFFER_LOAD_SHORT_D16::execute(GPUDynInstPtr gpuDynInst)
+    {
+        Wavefront *wf = gpuDynInst->wavefront();
+
+        if (gpuDynInst->exec_mask.none()) {
+            wf->decVMemInstsIssued();
+            return;
+        }
+
+        gpuDynInst->execUnitId = wf->execUnitId;
+        gpuDynInst->latency.init(gpuDynInst->computeUnit());
+        gpuDynInst->latency.set(gpuDynInst->computeUnit()->clockPeriod());
+
+        ConstVecOperandU32 addr0(gpuDynInst, extData.VADDR);
+        ConstVecOperandU32 addr1(gpuDynInst, extData.VADDR + 1);
+        ConstScalarOperandU128 rsrcDesc(gpuDynInst, extData.SRSRC * 4);
+        ConstScalarOperandU32 offset(gpuDynInst, extData.SOFFSET);
+
+        rsrcDesc.read();
+        offset.read();
+
+        int inst_offset = instData.OFFSET;
+
+        // For explanation of buffer addressing, see section 9.1.5 in:
+        // https://www.amd.com/content/dam/amd/en/documents/instinct-tech-docs/
+        //    instruction-set-architectures/
+        //    amd-instinct-mi300-cdna3-instruction-set-architecture.pdf
```

Comment:
I don't know we Matt will be able to get to this, but this was the last change requested, if someone things the comment is good enough to approve, as he already marked the other changes resolved.

Reply:
sorry, just looking now, at least from what I see above it seems fine.